-1

Tested at: 88a24ff46525eddaef418f8da2038473bca8470f

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-15261/14251/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found an error when building: 
 
 <pre>>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-07-26-2300/src/EventFilter/CastorRawToDigi/src/CastorCtdcUnpacker.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-07-26-2300/src/EventFilter/CastorRawToDigi/src/CastorMergerData.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-07-26-2300/src/EventFilter/CastorRawToDigi/src/CastorCTDCHeader.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-07-26-2300/src/EventFilter/CastorRawToDigi/src/CastorCtdcPacker.cc 
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-07-26-2300/src/EventFilter/HcalRawToDigi/plugins/HcalRawToDigi.cc: In member function 'virtual void HcalRawToDigi::produce(edm::Event&, const edm::EventSetup&)':
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-07-26-2300/src/EventFilter/HcalRawToDigi/plugins/HcalRawToDigi.cc:283:44: error: no matching function for call to 'std::auto_ptr<HcalUMNioDigi>::auto_ptr(HcalUMNioDigi&)'
     std::auto_ptr<HcalUMNioDigi> prod(umnio);
                                            ^
In file included from /cvmfs/cms-ib.cern.ch/2016-31/slc6_amd64_gcc530/external/gcc/5.3.0/include/c++/5.3.0/memory:85:0,
                 from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-07-26-2300/src/FWCore/Framework/interface/stream/implementors.h:22,
                 from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-07-26-2300/src/FWCore/Framework/interface/stream/AbilityToImplementor.h:25,
</pre>

* **Clang**:

I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 16 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>Selected class -> HcalUnpackerReport for ROOT: HcalUnpackerReport
Selected class -> HcalLaserDigi for ROOT: HcalLaserDigi
Selected class -> HcalUMNioDigi for ROOT: HcalUMNioDigi
>> Compiling LCG dictionary: tmp/slc6_amd64_gcc530/src/DataFormats/HcalDigi/src/DataFormatsHcalDigi/a/DataFormatsHcalDigi_xr.cc
1 warning generated.
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-07-26-2300/src/EventFilter/HcalRawToDigi/plugins/HcalRawToDigi.cc:283:34: error: no matching constructor for initialization of 'std::auto_ptr<HcalUMNioDigi>'
    std::auto_ptr<HcalUMNioDigi> prod(umnio);
                                 ^    ~~~~~
/cvmfs/cms-ib.cern.ch/2016-31/slc6_amd64_gcc530/external/gcc/5.3.0/lib/gcc/x86_64-pc-linux-gnu/5.3.0/../../../../include/c++/5.3.0/backward/auto_ptr.h:103:7: note: candidate constructor not viable: no known conversion from 'HcalUMNioDigi' to 'element_type *' (aka 'HcalUMNioDigi *') for 1st argument; take the address of the argument with &
      auto_ptr(element_type* __p = 0) throw() : _M_ptr(__p) { }
      ^
</pre>

--- a/EventFilter/HcalRawToDigi/plugins/HcalRawToDigi.cc
+++ b/EventFilter/HcalRawToDigi/plugins/HcalRawToDigi.cc
@@ -82,6 +82,7 @@ void HcalRawToDigi::fillDescriptions(edm::ConfigurationDescriptions& description
   desc.addUntracked<std::vector<int>>("FEDs", std::vector<int>());
   desc.addUntracked<bool>("UnpackZDC",true);
   desc.addUntracked<bool>("UnpackCalib",true);
+  desc.addUntracked<bool>("UnpackUMNio",true);
   desc.addUntracked<bool>("UnpackTTP",true);
   desc.addUntracked<bool>("silent",true);
   desc.addUntracked<bool>("ComplainEmptyData",false);

--- a/EventFilter/HcalRawToDigi/plugins/HcalRawToDigi.cc
+++ b/EventFilter/HcalRawToDigi/plugins/HcalRawToDigi.cc
@@ -281,7 +281,7 @@ void HcalRawToDigi::produce(edm::Event& e, const edm::EventSetup& es)
   /// umnio
   if (unpackUMNio_) {
     if(colls.umnio != 0) {
-      std::auto_ptr<HcalUMNioDigi> prod(colls.umnio);
+      std::auto_ptr<HcalUMNioDigi> prod = std::auto_ptr<HcalUMNioDigi>(new HcalUMNioDigi(umnio));
       e.put(prod);
     }
 

--- a/EventFilter/HcalRawToDigi/plugins/HcalRawToDigi.cc
+++ b/EventFilter/HcalRawToDigi/plugins/HcalRawToDigi.cc
@@ -281,7 +281,7 @@ void HcalRawToDigi::produce(edm::Event& e, const edm::EventSetup& es)
   /// umnio
   if (unpackUMNio_) {
     if(colls.umnio != 0) {
-      std::auto_ptr<HcalUMNioDigi> prod = std::auto_ptr<HcalUMNioDigi>(new HcalUMNioDigi(umnio));
+      std::auto_ptr<HcalUMNioDigi> prod(new HcalUMNioDigi(umnio));
       e.put(prod);
     }
 

--- a/EventFilter/HcalRawToDigi/src/HcalUnpacker.cc
+++ b/EventFilter/HcalRawToDigi/src/HcalUnpacker.cc
@@ -872,10 +872,6 @@ void HcalUnpacker::unpackUMNio(const FEDRawData& raw, int slot, Collections& col
   }
   const uint16_t* data = (const uint16_t*)(amc13->AMCPayload(namc));
   size_t nwords = amc13->AMCSize(namc) * ( sizeof(uint64_t) / sizeof(uint16_t) );
-  std::unique_ptr<HcalUMNioDigi> umnioDigi(new HcalUMNioDigi(data, nwords));
-  if(umnioDigi->invalid()) return;
+  *(colls.umnio) = HcalUMNioDigi(data, nwords);
   
-  colls.umnio = umnioDigi.get();
-  
-  return;
 }

